contracts/src/periphery/cert/router/EigenDACertVerifierRouter.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDACertVerifier} from "src/periphery/cert/interfaces/IEigenDACertVerifier.sol";
+import {IEigenDACertVerifierRouter} from "src/periphery/cert/interfaces/IEigenDACertVerifierRouter.sol";
+import {OwnableUpgradeable} from "lib/openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+
+contract EigenDACertVerifierRouter is IEigenDACertVerifierRouter, OwnableUpgradeable {
+    mapping(uint32 => address) public certVerifiers;
+    uint32[] public certVerifierRBNs;
+
+    event CertVerifierAdded(uint32 indexed rbn, address indexed certVerifier);
+
+    error RBNNotInFuture(uint32 rbn);
+    error RBNNotGreaterThanLast(uint32 rbn);
+    error InvalidCertLength();
+    error NoCertVerifierAvailable();
+    error NoCertVerifierFound(uint32 rbn);
+
+    /// IEigenDACertVerifierRouter ///
+
+    function checkDACert(bytes calldata certBytes) external view returns (uint8) {
+        return IEigenDACertVerifier(getCertVerifierAt(_getRBN(certBytes))).checkDACert(certBytes);
+    }
+
+    function getCertVerifierAt(uint32 rbn) public view returns (address) {
+        return certVerifiers[_findClosestRegisteredRBN(rbn)];
+    }
+
+    /// ADMIN ///
+
+    function initialize(address _initialOwner) external initializer {
+        _transferOwnership(_initialOwner);
+    }
+
+    function addCertVerifier(uint32 rbn, address certVerifier) external onlyOwner {
+        if (rbn <= block.number) {
+            revert RBNNotInFuture(rbn);
+        }
+        if (certVerifierRBNs.length > 0 && rbn <= certVerifierRBNs[certVerifierRBNs.length - 1]) {
+            revert RBNNotGreaterThanLast(rbn);
+        }
+        certVerifiers[rbn] = certVerifier;
+        certVerifierRBNs.push(rbn);
+        emit CertVerifierAdded(rbn, certVerifier);
+    }
+
+    /// INTERNAL ///
+
+    function _getRBN(bytes calldata certBytes) internal pure returns (uint32) {
+        if (certBytes.length < 36) {
+            revert InvalidCertLength();
+        }
+        return abi.decode(certBytes[32:36], (uint32));
+    }
+
+    /// @notice Given an RBN, find the closest RBN registered in this contract that is less than or equal to the given RBN.
+    /// @param referenceBlockNumber The reference block number to find the closest RBN for
+    /// @return closestRBN The closest RBN registered in this contract that is less than or equal to the given RBN.
+    function _findClosestRegisteredRBN(uint32 referenceBlockNumber) internal view returns (uint32) {
+        // It is assumed that the latest RBNs are the most likely to be used.
+        if (certVerifierRBNs.length == 0) {
+            revert NoCertVerifierAvailable();
+        }
+
+        uint256 rbnMaxIndex = certVerifierRBNs.length - 1; // cache to memory
+        for (uint256 i; i < certVerifierRBNs.length; i++) {
```

Comment:
if we're reverse iterating over the dynamic array then why not set the looping to instead iteratively decrement:
```
for (uint256 i = certVerifierRBNs.length; i >= 0; i--)
```
also worth noting that since we're doing a lookup on a ascending ordered array we could leverage a binary search to reduce operations here.

Reply:
Actually, I remembered why I didn't do this. It is troublesome to decrement in a for loop because the decrement happens after the check in the for loop definition, which makes a check for 0 impossible AFTER a decrement  to signal the end of the loop. Maybe we can use a while loop or something, but I'd rather not.... I think the solution is fine as is. It is relatively simple logic and efficient if we aren't planning on adding too many CertVerifiers.

Reply:
hmm curious if there'd be security callouts with this though given a its a dynamic array and does introduce a DoS vector. It's also likely that in the rollup domain that the role who updates the `Router` could differ from the one who holds the admin proxy.  If it better future proofs the contract and eliminates a security vulnerability then why not include it? Otherwise we could enforce some size limit on the array or some form of expiration for historical entries.

> I don't imagine it will be common to prove blobs too far in the past.
Sadly there are customers who archive all EigenDA blobs for their rollups behind proxy on S3 to provide safe/final head syncing from genesis without needing an archival execution layer node.